virtualization-and-containers/material/vagrant_slides.md

```diff
@@ -0,0 +1,197 @@
+---
+type: slide
+slideOptions:
+  transition: slide
+  width: 1400
+  height: 900
+  margin: 0.1
+---
+
+<style>
+  .reveal strong {
+  font-weight: bold;
+    color: orange;
+  }
+  .reveal p {
+    text-align: left;
+  }
+  .reveal section h1 {
+    color: orange;
+  }
+  .reveal section h2 {
+    color: orange;
+  }
+</style>
+
+# Vagrant
+
+---
+
+## Introduction
+
+- Remember:
+  > If you can't git diff a file format, it's broken.
+- Infrastructure as code
+  > Vagrant is a tool for building and managing virtual machine environments in a single workflow.
+  [https://www.vagrantup.com/intro](https://www.vagrantup.com/intro)
+- Initial release 2010 by Mitchell Hashimoto, now [HashiCorp](https://www.hashicorp.com/)
+- HashiCorp develops open-source DevOps/Cloud tools like [Vagrant](https://www.vagrantup.com/), [TerraForm](https://www.terraform.io/)...
+
+---
+
+## What for?
+
+- Developing software
+  - Set up consistent development environment
+  - Share environment with others
+- Testing software, workflows...
+  - Disposable environments
+  - Consistent workflows
+- More general
+  - Easy way to setup virtual machine
+
+---
+
+## Advantages
+
+- Strong focus on workflow consistency
+- Avoid usage of (VirtualBox) CLI
+- Store in Git-friendly format
+- For us:
+  - Management of VirtualBox VMs (testing, developing...)
+  - Sharing of VMs (debugging, workshops...)
+
+---
+
+## Building Block
+
+- Provisioners
+  - Tool for configuring the boxes (install software on machine)
+  - Examples: shell scripts, Chef, Puppet..
+- Providers
+  - "Backend" the box is based on
+  - Examples: VirtualBox, Hyper-V, AWS, Docker...
```

Comment:
Does this slide really belong here? Only reading it, the connection to the surrounding slides is unclear.

Reply:
I agree. I moved it just after the introduction.